samza-core/src/main/java/org/apache/samza/execution/StreamEdge.java

```diff
@@ -125,6 +125,7 @@ Config generateConfig() {
     if (isIntermediate()) {
       config.put(String.format(StreamConfig.IS_INTERMEDIATE_FOR_STREAM_ID(), spec.getId()), "true");
       config.put(String.format(StreamConfig.CONSUMER_OFFSET_DEFAULT_FOR_STREAM_ID(), spec.getId()), "oldest");
+      config.put(String.format(StreamConfig.PRIORITY_FOR_STREAM_ID(), spec.getId()), "1");
```

Comment:
Consider adding an assert to https://github.com/apache/samza/blob/master/samza-core/src/test/java/org/apache/samza/execution/TestStreamEdge.java as well if possible?